DomainContent/Zaru/applause-app/applause-app.js

```diff
@@ -0,0 +1,407 @@
+    }
+
+    function checkHandsDistance() {
+
+        var handPositionR = MyAvatar.getJointPosition('RightHand');
+        var handPositionL = MyAvatar.getJointPosition('LeftHand');
+        var handRotationR = MyAvatar.getRightPalmRotation();
+        var handRotationL = MyAvatar.getLeftPalmRotation();
+        var oldRotationR = previousHandOrientations[0];
+        var oldRotationL = previousHandOrientations[1];
+
+        if (Vec3.distance(handPositionL, handPositionR) <= handProximityDistance) {
+            if (!handsStretched) {
+                makeOpenPalm();
+                handsStretched = true;
+            }
+        } else {
+            if (handsStretched) {
+                clearJoints();
+                handsStretched = false;
+            }
+        }
+        if ((Vec3.distance(handPositionL, handPositionR) <= handProximityDistance) 
+            && (compareRotations(oldRotationR, handRotationR) 
+            || compareRotations(oldRotationL, handRotationL)) && applauseEnabled) {
+            playSingleClap();
+            previousHandOrientations = [];
+            previousHandOrientations.push(handRotationR);
+            previousHandOrientations.push(handRotationL);
+        }
+    }
+    // Tablet Handlers
+    function onClicked() {
+        if (open) {
+            tablet.gotoHomeScreen();
+        } else {
+            tablet.gotoWebScreen(appPage);
+        }
+    }
+    function onWebEventReceived(event) {
+        if (typeof event === 'string') {
+            event = JSON.parse(event);
+        }
+        switch (event.type) {
+            case 'applause-app-ready':
+                var webEvent = {
+                    type: 'setup',
+                    enabled : Settings.getValue(CAN_APPLAUD_SETTING),
+                    particles: Settings.getValue(SHOULD_DISPLAY_PARTICLES_APPLAUSE, true),
+                    HMD : HMD.active
+                };
+                tablet.emitScriptEvent(JSON.stringify(webEvent));
+                break;
+            
+            case 'clap':
+                if (applauseEnabled) {
+                    playSingleClap();
+                }
+                break;
+
+            case 'applause-enabled':
+                applauseEnabled = event.value;
+                Settings.setValue(CAN_APPLAUD_SETTING, applauseEnabled);
+                if (applauseEnabled) {
+                    setup();
+                } else {
+                    disableApplause();
+                }
+                break;
+
+            case 'change-particles':
+                shouldUseParticles = event.value;
+                Settings.setValue(SHOULD_DISPLAY_PARTICLES_APPLAUSE, event.value);
+                break;
+            
+            default: 
+                break;
+        }
+    }
+
+    var setup = function() {
+        if (HMD.active) {
+            removeDesktopOverlay();
+            Script.update.connect(checkHandsDistance);
```

Comment:
Doesn't this design mean that `checkHandsDistance` can get connected more than once?

Reply:
It's disconnected when desktop mode becomes active - So I'm not positive how it would get connected multiple times before a disconnect was called. Is there a way to check that something isn't already connected to a signal before connecting it?

Reply:
I think I'm missing something; I don't see where it's disconnected when desktop mode becomes active. Could you please point me to that?

Regarding your second question - in [wallet.js](https://github.com/highfidelity/hifi/blob/master/scripts/system/commerce/wallet.js#L596), we use a couple of `bool`s (`isWired` and `isUpdateOverlaysWired`) to keep the state of the connections such that we don't double-connect or double-disconnect.

Reply:
Ack - you're right, it's not. I misread it in another function.